spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/DefaultAmqpHeaderMapper.java

```diff
@@ -405,4 +411,50 @@ private void addConsumerMetadata(MessageProperties messageProperties, Map<String
 		}
 	}
 
+	public static DefaultAmqpHeaderMapper inboundMapper() {
+		DefaultAmqpHeaderMapper mapper = new DefaultAmqpHeaderMapper();
+		mapper.setRequestHeaderNames(inboundRequestHeaders());
+		mapper.setReplyHeaderNames(inboundReplyHeaders());
+		return mapper;
+	}
+
+	public static DefaultAmqpHeaderMapper outboundMapper() {
+		DefaultAmqpHeaderMapper mapper = new DefaultAmqpHeaderMapper();
+		mapper.setRequestHeaderNames(outboundRequestHeaders());
+		mapper.setReplyHeaderNames(outboundReplyHeaders());
+		return mapper;
+	}
+
+	/**
+	 * @return the default request headers for an inbound mapper.
+	 */
+	public static String[] inboundRequestHeaders() {
+		return safeInboundHeaders();
+	}
+
+	/**
+	 * @return the default reply headers for an inbound mapper.
+	 */
+	public static String[] inboundReplyHeaders() {
+		return new String[] { "*" };
+	}
+
+	/**
+	 * @return the default request headers for an outbound mapper.
+	 */
+	public static String[] outboundRequestHeaders() {
+		return new String[] { "*" };
+	}
+
+	/**
+	 * @return the default reply headers for an outbound mapper.
+	 */
+	public static String[] outboundReplyHeaders() {
+		return safeInboundHeaders();
```

Comment:
Yes, consider `ibgw -> obgw`; without this, the reply headers will be propagated.